model/dyn_core.F90

```diff
@@ -1234,6 +1253,156 @@ subroutine dyn_core(npx, npy, npz, ng, sphum, nq, bdt, n_map, n_split, zvir, cp,
 
     endif
 
+#ifdef MOLECULAR_DIFFUSION
```

Comment:
I really think this all needs to be put in a separate subroutine instead of leaving it in-line in dyn_core()/

Reply:
They are moved in a subroutine now.